src/rGitignore.ts

```diff
@@ -1,32 +1,77 @@
 'use strict';
 
 import { writeFile } from 'fs-extra';
+import { existsSync } from 'fs';
 import { join } from 'path';
 import { window, workspace } from 'vscode';
-// From "https://github.com/github/gitignore/raw/master/R.gitignore"
-const ignoreFiles = ['.Rhistory',
-                     '.Rapp.history',
-                     '.RData',
-                     '*-Ex.R',
-                     '/*.tar.gz',
-                     '/*.Rcheck/',
-                     '.Rproj.user/',
-                     'vignettes/*.html',
-                     'vignettes/*.pdf',
-                     '.httr-oauth',
-                     '/*_cache/',
-                     '/cache/',
-                     '*.utf8.md',
-                     '*.knit.md',
-                     'rsconnect/'].join('\n');
+// .gitignore template from "https://github.com/github/gitignore/blob/main/R.gitignore"
+const ignoreFiles = ['# History files',
+    '.Rhistory',
+    '.Rapp.history',
+    '',
+    '# Session Data files',
+    '.RData',
+    '.RDataTmp',
+    '',
+    '# User-specific files',
+    '.Ruserdata',
+    '',
+    '# Example code in package build process',
+    '*-Ex.R',
+    '',
+    '# Output files from R CMD build',
+    '/*.tar.gz',
+    '',
+    '# Output files from R CMD check',
+    '/*.Rcheck/',
+    '',
+    '# RStudio files',
+    '.Rproj.user/',
+    '',
+    '# produced vignettes',
+    'vignettes/*.html',
+    'vignettes/*.pdf',
+    '',
+    '# OAuth2 token, see https://github.com/hadley/httr/releases/tag/v0.3',
+    '.httr-oauth',
+    '',
+    '# knitr and R markdown default cache directories',
+    '*_cache/',
+    '/cache/',
+    '',
+    '# Temporary files created by R markdown',
+    '*.utf8.md',
+    '*.knit.md',
+    '',
+    '# R Environment Variables',
+    '.Renviron',
+    '',
+    '# pkgdown site',
+    'docs/',
+    '',
+    '# translation temp files',
+    'po/*~',
+    '',
+    '# RStudio Connect folder',
+    'rsconnect/',
+    ''].join('\n');
 
-export function createGitignore(): void {
+export async function createGitignore(): Promise<void> {
     if (workspace.workspaceFolders[0].uri.path === undefined) {
         void window.showWarningMessage('Please open workspace to create .gitignore');
 
         return;
     }
     const ignorePath = join(workspace.workspaceFolders[0].uri.path, '.gitignore');
```

Comment:
Just notice that the usage of `workspace.workspaceFolders[0].uri.path` here does not work with no workspace case, nor does it work with multi-root workspace. Do you want to fix it here?

Reply:
Sorry, I didn't know the proper way to fix this.

Reply:
Never mind, I'll fix this later in a separate PR.